pkg/apis/core/validation/validation.go

```diff
@@ -1966,18 +1966,53 @@ func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *core.PersistentVolumeCl
 	if utilfeature.DefaultFeatureGate.Enabled(features.ExpandPersistentVolumes) {
 		// lets make sure storage values are same.
 		if newPvc.Status.Phase == core.ClaimBound && newPvcClone.Spec.Resources.Requests != nil {
-			newPvcClone.Spec.Resources.Requests["storage"] = oldPvc.Spec.Resources.Requests["storage"]
+			newPvcClone.Spec.Resources.Requests[core.ResourceStorage] = oldPvc.Spec.Resources.Requests[core.ResourceStorage]
+		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) {
+			// set new PVC's allocatedResources to be old pvc allocatedResources so as
+			// even if they are nil validations could pass
+			newPvcClone.Spec.AllocatedResources = oldPvc.Spec.AllocatedResources
 		}
 
 		oldSize := oldPvc.Spec.Resources.Requests["storage"]
 		newSize := newPvc.Spec.Resources.Requests["storage"]
 
+		statusSize := oldPvc.Status.Capacity["storage"]
+
 		if !apiequality.Semantic.DeepEqual(newPvcClone.Spec, oldPvcClone.Spec) {
 			specDiff := diff.ObjectDiff(newPvcClone.Spec, oldPvcClone.Spec)
 			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), fmt.Sprintf("spec is immutable after creation except resources.requests for bound claims\n%v", specDiff)))
 		}
+
+		var oldAllocatedSize resource.Quantity
+		var newAllocatedSize resource.Quantity
+
+		if oldPvc.Spec.AllocatedResources != nil {
+			oldAllocatedSize = oldPvc.Spec.AllocatedResources.Requests[core.ResourceStorage]
+		}
+		if newPvc.Spec.AllocatedResources != nil {
```

Comment:
is the old valid value in this map `core.ResourceStorage`? if so, can we enforce that via validation to prevent accidentally storing more data than we mean to?

Reply:
I do not understand this. can you reword it please?